src/aat/java/uk/gov/hmcts/ccd/datastore/tests/functional/elasticsearch/ElasticsearchCaseSearchSecurityTest.java

```diff
@@ -0,0 +1,177 @@
+        void shouldNotReturnCaseFieldForLowerSecurityClassification() {
+            ValidatableResponse response = searchCaseAndAssertCaseReference(asPrivateCaseworker(false),
+                                                                            ES_FIELD_CASE_REFERENCE,
+                                                                            caseRefCaseFieldSecurityQuery);
+            response.body("cases[0].case_data", not(hasKey(ES_FIELD_EMAIL_ID)));
+        }
+
+        //@Test
+        @DisplayName("should NOT return the case field where user role does not have read access on the field")
+        void shouldNotReturnCaseFieldForNoReadAccess() {
+            //TODO
+        }
+
+    }
+
+    @Nested
+    @DisplayName("User access security")
+    class UserAccessSecurity {
+
+        @Test
+        @DisplayName("should return the case for a solicitor role if granted access to the case")
+        void shouldReturnCase() {
+            //TODO
+        }
+
+        //@Test
+        @DisplayName("should NOT return the case for a solicitor role if not granted access to the case")
+        void shouldNotReturnCase() {
+            //TODO
+        }
+
+    }
+
+    private ValidatableResponse searchCaseAndAssertCaseReference(Supplier<RequestSpecification> asUser, String field, Object value) {
+        ValidatableResponse response = searchCase(asUser, field, value);
+        response.body("cases.size()", is(1));
+        response.body("cases[0]." + CASE_ID, is(value));
+
+        return response;
+    }
+
+    private void searchCaseAndAssertCaseNotReturned(Supplier<RequestSpecification> asUser, String field, Object value) {
+        ValidatableResponse response = searchCase(asUser, field, value);
+        response.body("cases.size()", is(0));
+    }
+
+    private ValidatableResponse searchCase(Supplier<RequestSpecification> asUser, String field, Object value) {
+        String jsonSearchRequest = createExactMatchSearchRequest(field, value);
+        return searchCase(asUser, jsonSearchRequest);
+    }
+
+    private String createExactMatchSearchRequest(String field, Object value) {
+        return "{"
+            + "  \"query\": {"
+            + "    \"match\": {"
+            + "    \"" + field + "\" : \"" + String.valueOf(value) + "\""
+            + "    }"
+            + "  }"
+            + "}";
+    }
+
+    @AfterAll
+    void cleanUp() {
+        deleteIndexAndAlias();
+    }
+
+    private void createCases() {
+        caseRefCaseTypeSecurityClassificationQuery = createCase(asPrivateCaseworker(true), AATCaseBuilder.EmptyCase.build());
+        caseRefCaseStateSecurityQuery = createCaseAndProgressState(asPrivateCaseworker(true));
+        caseRefCaseFieldSecurityQuery = createCase(asRestrictedCaseworker(true),
+                                                   AATCaseType.CaseData.builder().emailField(EMAIL_ID_VALUE).build());
+
+        // wait until logstash reads the case data
```

Comment:
can we introduce a method called `waitIndexingCompletex` and remove this comment?

Reply:
Done